### PR TITLE
Fixed changing device name

### PR DESCRIPTION
**Pull request by igor-sirotin**
Fixes https://github.com/status-im/status-desktop/issues/9584
Requires https://github.com/status-im/status-go/pull/3294

### What does the PR do

Now it's possible to change device name :)

https://user-images.githubusercontent.com/25482501/225600994-24f7d13c-8049-4aff-bd1b-4e878a51bdce.mov

### CODE REVIEW

File: src/app/modules/main/profile_section/devices/model.nim [hunk 1]
```diff
@@ -122,5 +122,15 @@ QtObject:
     self.items[i].installation = installation
     self.dataChanged(first, last, @[])
 
+  proc updateItemName*(self: Model, installationId: string, name: string) =
+    var i = self.findIndexByInstallationId(installationId)
+    if(i == -1):
+      return
+
+    let first = self.createIndex(i, 0, nil)
+    let last = self.createIndex(i, 0, nil)
```
Comment (caybro) [inline src/app/modules/main/profile_section/devices/model.nim:131]: Those are the same index right? And it has to be defer-deleted

File: ui/app/AppLayouts/Profile/popups/SyncDeviceCustomizationPopup.qml [hunk 2]
```diff
@@ -50,8 +58,7 @@ StatusDialog {
                 text: qsTr("Done")
                 enabled: nameInput.text !== ""
                 onClicked : {
-                    root.devicesStore.setName(nameInput.text.trim())
-                    root.close();
+                    d.saveNewName()
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/popups/SyncDeviceCustomizationPopup.qml:61]: `root.close()` removed on purpose?

**Review by caybro — COMMENTED** [review]
Almost there :)

**Review by caybro — APPROVED** [review]

File: src/app_service/service/devices/service.nim [hunk 3]
```diff
@@ -117,10 +123,15 @@ QtObject:
       let errDesription = e.msg
       error "error: ", errDesription
 
-  proc setDeviceName*(self: Service, name: string) =
-    let installationId = self.settingsService.getInstallationId()
+  proc setInstallationName*(self: Service, installationId: string, name: string) =
     # Once we get more info from `status-go` we may emit success/failed signal from here.
-    discard status_installations.setInstallationMetadata(installationId, name, hostOs)
+    let response = status_installations.setInstallationName(installationId, name)
```
Comment (mprakhov) [inline src/app_service/service/devices/service.nim:128]: shouldn't it be in the try block?

File: src/app_service/service/devices/service.nim [hunk 4]
```diff
@@ -117,10 +123,15 @@ QtObject:
...
+    if response.error != nil:
+      let e = Json.decode($response.error, RpcError)
+      error "error: ", errorDescription = e.message
+      discard
```
Comment (mprakhov) [inline src/app_service/service/devices/service.nim:132]: should we do a return here?
Or discard won't execute code after?

**Comment by igor-sirotin** [comment]
@mprakhov thanks for review! 
You're right with both points, fixed 🙂

**Review by mprakhov — APPROVED** [review]
lgtm

**Pull request merged**